Source/Data/HAEntity+CompressedEntity.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+extension HAEntity {
+    func updatedEntity(compressedEntityState: CompressedEntityState) -> HAEntity? {
```

Comment:
why not make this mutating instead? i feel like it reads easier and since it's a struct there's no reason to worry about mutating underneath somebody using the old one

e.g., this feels good:

```
mutating func update(from state: CompressedEntityState)
mutating func add(_ state: CompressedEntityState)
mutating func subtract(_ state: CompressedEntityState)
```